airflow/contrib/sensors/jira_sensor.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from airflow.contrib.operators.jira_operator import JiraOperator
+from airflow.operators.sensors import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class JiraSensor(BaseSensorOperator):
+    """
+    Monitors a jira ticket for any change.
+
+    :param jira_conn_id: reference to a pre-defined Jira Connection
+    :type jira_conn_id: str
+    :param method_name: method name from jira-python-sdk to be execute
+    :type method_name: str
+    :param method_params: parameters for the method method_name
+    :type method_params: dict
+    :param result_processor: function that return boolean and act as a sensor response
+    :type result_processor: function
+    """
+
+    @apply_defaults
+    def __init__(self, jira_conn_id='jira_default',
+                 method_name=None,
+                 method_params=None,
+                 result_processor=None,
+                 *args,
+                 **kwargs):
+        super(JiraSensor, self).__init__(*args, **kwargs)
+        self.jira_conn_id = jira_conn_id
+        self.result_processor = result_processor
+        self.method_name = method_name
+        self.method_params = method_params
+
+    def poke(self, context):
+        logging.debug('Jira Sensor poking for : %s', self.method_name)
+
+        jira_operator = JiraOperator(task_id=self.task_id,
```

Comment:
Perhaps this `JiraOperator` could be created in `__init__` to avoid having to instantiate it every time `poke()` is called?